module-system/sov-state/src/prover_storage.rs

```diff
@@ -67,6 +74,12 @@ impl<S: MerkleProofSpec> Storage for ProverStorage<S> {
         val
     }
 
+    fn get_state_root(&self, _witness: &Self::Witness) -> [u8; 32] {
+        self.get_root_hash(self.db.get_next_version() - 1)
+            .map(|root| root.0)
+            .unwrap()
```

Comment:
Can this method panic?

Reply:
That's a good question: for me if there is no state root this function should panic indeed because the state root should always be set. But the get_root_hash function returns a result so I guess this can return a result as well

Reply:
Ok, then I would say replacing unwrap with `expect` for faster debugging would be enoug

Reply:
I added an error to this method